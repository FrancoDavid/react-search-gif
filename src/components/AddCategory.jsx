import React, { useState } from 'react';
import PropType from 'prop-types';
import '../App.css';

const AddCategory = (props) => {

    const [inputValue, setInputValue] = useState('');

    const handlerInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handlerSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 0) {
            props.setCategorias((cats) => [...cats, inputValue]);
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handlerSubmit}>
            <input
                className="Input-search"
                type="text"
                placeholder="Busca aquÃ­ tu gif en react app..."
                onChange={handlerInputChange}
                value={inputValue}/>
        </form>
    );
}

AddCategory.propType = {
    setCategorias: PropType.func.isRequired
};
 
export default AddCategory;