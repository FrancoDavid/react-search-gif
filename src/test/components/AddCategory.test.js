import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';
import React from 'react';

describe('Test add category component', () => {
    const setCategorias = jest.fn();
    let component = shallow(<AddCategory setCategorias={setCategorias}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        component = shallow(<AddCategory setCategorias={setCategorias}/>);
    });

    test('show correctly', () => {
        expect(component).toMatchSnapshot();
    });

    // test('input changes correctly', () => {
    //     const input = component.find('input');
    //     const value = 'labios de ruby';
    //     input.simulate('change', {target: value});
    //     console.log(input.props());
    //     // expect(input.prop('value')).toBe(value);
    // });

    test('validate form not submit', () => {
        component.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategorias).not.toHaveBeenCalled();
    });


    test('set categories should be called and value input should be clean', () => {
        const value = 'Hola mundo';
        component.find('input').simulate('change', {target: {value}});
        component.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategorias).toHaveBeenCalled();
        expect(component.find('input').prop('value')).toBe('');
    });
});
