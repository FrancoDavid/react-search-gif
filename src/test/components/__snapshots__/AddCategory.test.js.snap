// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test add category component show correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="Input-search"
    onChange={[Function]}
    placeholder="Busca aquÃ­ tu gif en react app..."
    type="text"
    value=""
  />
</form>
`;
